src/ir/export-utils.cpp

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright 2023 WebAssembly Community Group participants
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "ir/export-utils.h"
+#include "wasm.h"
+
+namespace wasm::ExportUtils {
+
+namespace {
+
+// Returns a vector of all the things that are exported of a particular kind,
+// given that kind and a way to look them up on the module.
+template<typename T, ExternalKind K, typename G>
```

Comment:
It would be possible to infer `G` from `K` or vice versa, but if this will always be wrapped in a non-template function, then it wouldn't be worth it.

Reply:
Yeah, I think maybe it's not worth it here. But adding a `getKindFromType` function does seem possible in principle (maybe we already have one and I'm not fully paged in yet as well? :smile: but I can't seem to see one)